src/main/java/org/elasticsearch/action/admin/indices/template/put/TransportPutIndexTemplateAction.java

```diff
@@ -64,7 +64,7 @@ protected PutIndexTemplateResponse newResponse() {
 
     @Override
     protected ClusterBlockException checkBlock(PutIndexTemplateRequest request, ClusterState state) {
-        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA, "");
+        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA_WRITE, "");
```

Comment:
not related to this change, but I get confused by these calls with empty index. Will they ever return a block? Seems like we should dig here on another PR.